docs/rules/order.md

```diff
@@ -77,21 +79,24 @@ This rule supports the following options:
 
 ### `groups: [array]`:
 
-How groups are defined, and the order to respect. `groups` must be an array of `string` or [`string`]. The only allowed `string`s are: `"builtin"`, `"external"`, `"internal"`, `"unknown"`, `"parent"`, `"sibling"`, `"index"`. The enforced order is the same as the order of each element in a group. Omitted types are implicitly grouped together as the last element. Example:
+How groups are defined, and the order to respect. `groups` must be an array of `string` or [`string`]. The only allowed `string`s are:
+`"builtin"`, `"external"`, `"internal"`, `"unknown"`, `"parent"`, `"sibling"`, `"index"`, `"object"`.
+The enforced order is the same as the order of each element in a group. Omitted types are implicitly grouped together as the last element. Example:
 ```js
 [
   'builtin', // Built-in types are first
   ['sibling', 'parent'], // Then sibling and parent types. They can be mingled together
   'index', // Then the index file
+  'object',
   // Then the rest: internal and external type
 ]
 ```
-The default value is `["builtin", "external", "parent", "sibling", "index"]`.
+The default value is `["builtin", "external", "parent", "sibling", "index", "object"]`.
```

Comment:
generally it's a breaking change to alter the default value here. can you help me understand why this wouldn't be breaking?

Reply:
Oh... Sorry, my interpretation of "breaking change" was a change of the API 😅

If changing the default value of a setting is a breaking change, then this indeed is one.

Reply:
What happens now with the ordering of object imports, versus what happens after this change?

Reply:
Before #1785, `import [...] = Object.literal;`s were ignored as well as `import [...] = require("module-name");`.
After #1785 `import [...] = require([...]);`s were included into the sorting while object-imports caused an error cause of this piece of code:
https://github.com/benmosher/eslint-plugin-import/blob/62b554bc5f5024695b40d8701fbd8267fb4eb73f/src/rules/order.js#L614-L616

After this PR, #1823, object-imports are included into the sorting, having their own group called `object`. Additionally, object-imports are then sorted according to their object-literal:

```ts
/* eslint import/order: ["error", {"alphabetize": {"order": "asc"}}] */
import path = require("path");
import { ESLint } from "eslint";
import Options = ESLint.Options; // This will fail, because "ESLint.Options" must be imported after "ESLint.LintResult"
import LintResult = ESLint.LintResult;
```

Reply:
That means that valid code before would suddenly start being invalid, and an autofix would suddenly move code.

This means that we can't add this category to the defaults at this time.

Reply:
But wouldn't removing this from the `groups` cause it to be forced to be at the end anyway?
At least if I understand this correctly:
https://github.com/benmosher/eslint-plugin-import/blame/13d2e10fa42e5a978fa00a52a1b10553913d3c16/docs/rules/order.md#L84

Reply:
Should I remove `object` from the default value of the `groups` setting?

Reply:
I would expect that with it not in "groups" they should be ignored entirely.

Reply:
The current behavior is that all of the missing groups are treated as the same - meaning, not sorted within themselves, just mushed together. By adding it, you're forcing `object` to go *before* all the other missing ones.

so yes, it should be removed.

Reply:
Aaaah got it - thanks for the elaboration :D

Reply:
There you go, I just reverted the mutation of the default settings 😄